pxf/pxf-hive/src/main/java/org/apache/hawq/pxf/plugins/hive/HiveORCSerdeResolver.java

```diff
@@ -102,11 +104,19 @@ void initSerde(InputData input) throws Exception {
 
         StringBuilder columnNames = new StringBuilder(numberOfDataColumns * 2); // column + delimiter
         StringBuilder columnTypes = new StringBuilder(numberOfDataColumns * 2); // column + delimiter
+        String[] tmp = typesString.split(":");
+        String[] hiveColTypes = new String[numberOfDataColumns];
+        parseColTypes(tmp, hiveColTypes);
+
         String delim = ",";
         for (int i = 0; i < numberOfDataColumns; i++) {
             ColumnDescriptor column = input.getColumn(i);
             String columnName = column.columnName();
             String columnType = HiveUtilities.toCompatibleHiveType(DataType.get(column.columnTypeCode()), column.columnTypeModifiers());
+            //Complex Types will have a mismatch between Hive and Hawq type
```

Comment:
The only issue I have with that approach is that whether a column is a complex type is table specific so HiveUtilities.toCompatibleHiveType would need to also take the Hive column information as an additional parameter since Hawq type text could either be a Hive string or any Hive complex type.

I think it does make sense to put the below logic into a utility method